Modules/posixmodule.c

```diff
@@ -5026,45 +5112,73 @@ os__path_islink_impl(PyObject *module, PyObject *path)
     }
 
     Py_BEGIN_ALLOW_THREADS
-    if (_path.fd != -1) {
-        hfile = _Py_get_osfhandle_noraise(_path.fd);
-        close_file = FALSE;
+    if (_path.wide) {    
+        if (_Py_GetFileInformationByName(_path.wide, FileStatBasicByNameInfo,
+                                         &statInfo, sizeof(statInfo))) {
+            if (// Cannot use fast path for reparse points ...
+                !(statInfo.FileAttributes & FILE_ATTRIBUTE_REPARSE_POINT)
+                // ... unless it's a name surrogate (symlink)
+                || IsReparseTagNameSurrogate(statInfo.ReparseTag)
+            ) {
+                slow_path = FALSE;
+                result = (statInfo.ReparseTag == IO_REPARSE_TAG_SYMLINK);
+            }
```

Comment:
In this case a successful result from `GetFileInformationByName()` suffices. 

```suggestion
            slow_path = FALSE;
            if (statInfo.FileAttributes & FILE_ATTRIBUTE_REPARSE_POINT) {
                result = (statInfo.ReparseTag == IO_REPARSE_TAG_SYMLINK);
            }
            else {
                result = FALSE;
            }
```